modules/apps/27-interchain-accounts/keeper/account.go

```diff
@@ -56,3 +56,16 @@ func (k Keeper) RegisterInterchainAccount(ctx sdk.Context, accAddr sdk.AccAddres
 	k.accountKeeper.SetAccount(ctx, interchainAccount)
 	k.SetInterchainAccountAddress(ctx, controllerPortID, interchainAccount.Address)
 }
+
+// Opens a new channel on a particular port given a connection
+// This is a helper function to open a new channel
+// in case of a channel closing and the controller chain needs to regain access to an interchain account on the host chain
+func (k Keeper) InitChannel(ctx sdk.Context, portID, connectionID string) error {
+	msg := channeltypes.NewMsgChannelOpenInit(portID, types.VersionPrefix, channeltypes.ORDERED, []string{connectionID}, types.PortID, types.ModuleName)
```

Comment:
I was initially considering doing a check to see if there is an active channel currently set and returning an err if so but on second thought I liked the idea of this being more of an  'admin' function capable of overriding the active channel and just creating a new channel (and thus setting a new active channel). 

I'm happy to add in this check and return an error if we think it's necessary. I'm leaning more towards putting this check in, but would like to hear what others think.

Reply:
the onChanOpenInit function checks this anyway, so it's sort of a moot point actually